client/src/App.tsx

```diff
@@ -16,6 +17,10 @@ const router = createBrowserRouter([
     path: '/signup',
     element: <SignUp />,
   },
+  {
+    path: '/menu',
+    element: <Menu />,
```

Comment:
You have an error here saying that `type of sting | Element is not assignable to type ReactElement<any, any>`.
This error happens because you returned a string as a loader and a string when you find errors, modify this to tags or MUI components.